src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderStrips.java

```diff
@@ -285,4 +324,57 @@ public BufferedImage readImageData(final Rectangle subImage)
                 subImage.height);
     }
 
+    @Override
+    public TiffRasterData readRasterData(Rectangle subImage)
+        throws ImageReadException, IOException {
+
+        int xRaster;
+        int yRaster;
+        int rasterWidth;
+        int rasterHeight;
+        if (subImage != null) {
+            xRaster = subImage.x;
+            yRaster = subImage.y;
+            rasterWidth = subImage.width;
+            rasterHeight = subImage.height;
+        } else {
+            xRaster = 0;
+            yRaster = 0;
+            rasterWidth = width;
+            rasterHeight = height;
+        }
+        float[] rasterData = new float[rasterWidth * rasterHeight];
+
+        // the legacy code is optimized to the reading of whole
+        // strips (except for the last strip in the image, which can
+        // be a partial).  So create a working image with compatible
+        // dimensions and read that.  Later on, the working image
+        // will be sub-imaged to the proper size.
+        // strip0 and strip1 give the indices of the strips containing
+        // the first and last rows of pixels in the subimage
+        final int strip0 = yRaster / rowsPerStrip;
```

Comment:
Any change—even if remote—that `rowsPerStrip` could be zero?

One of the reasons why 1.0-alpha1 took so long (couple years after the work started?) was largely due to CVE's created for sanselan.

One of the CVE, if I am not mistaken, was about a division by zero (or another sort of arithmetic error?), that caused an exception and some infinite loop or large memory allocation (allowing users to perform DDOS).

If it's never zero, then no need to worry. Otherwise, if there's any risk, better set a quick validation somewhere (here or constructor).

Reply:
I would think that the only way rowsPerStrip would be zero is if the source file was corrupted.  TIFF files can include directories that don't have images (just metadata).  But in such a case, I think the code would never get here.

I could add tests for this condition, but if it never happens it will just bring down the Coveralls score.  So what I will do is to go through the code and see if there is logic some place to detect a corrupt image and through a meaningful exception if found.

Reply:
>So what I will do is to go through the code and see if there is logic some place to detect a corrupt image and through a meaningful exception if found.

Excellent!

The CVE's that blocked our release for a long time, had JPEG files that were randomly generated with a fuzzer. I think one of the images was corrupted/invalid format, while the other rendered something on GIMP and exiftool parsed it, but still had some invalid data.

The issue was really that if a user used commons-imaging on a webapp, parsing images like that could break their app. So better code defensive if we can :+1: